### PR TITLE
spreadsheet sort enhancements

### LINKED ISSUE #2152: 1817 sort spreadsheet by loans/shorts
Would be nice to be able to sort by size

### LINKED ISSUE #3799: 18Mag sort by player doesn't sort by minor
https://18xx.games/game/26627?finished=13#spreadsheet

When I sort the spreadsheet by player, the minor corporations are not sorted. It would be nice if the * marking ownership of a minor was included in the sorting, so that all the minors owned by (in this example) Excalabur appeared in a group.

Screen Shot 2021-02-11 at 2 06 50 PM

### PR DESCRIPTION
**Pull request by yzemaze**
- corporation size added (only shown when there are different sizes of corps available) – fixes #2152 (body text mentions corp size)
- include minors/president-status when sorting player column – fixes #3799
- shares == 0 => hidden
- operating order == 0 => hidden
- operated => grayed out instead of *; alignment changed to right
- shorts sorted between longs and 0
- sort order toggles only when currently sorted column is sorted again – instead of when changing sort column
- sort order indicator (arrow) moved to left of column header, because it gets hidden too often due to width limit of player columns
- or_history right aligned & a tad more padding

tl;dr
image
image

### CODE REVIEW

**Comment by crericha** [comment]
Would it make sense to have the S, M, L company size string be defined by the game class?

Example strings could be:

1817: 2, 5, 10
18CZ: S, M, L
1861/7: Minor, Major

File: assets/app/view/game/spreadsheet.rb [hunk 1]
```diff
@@ -177,7 +176,11 @@ def render_titles
           extra << h(:th, render_sort_link('Buying Power', :buying_power))
           extra << h(:th, render_sort_link('Interest Due', :interest))
         end
+        if (@diff_corp_sizes = @game.all_corporations.map { |c| @game.corporation_size(c) }.uniq != [:small])
```
Comment (michaeljb) [inline assets/app/view/game/spreadsheet.rb:179]: Just one nitpick, this iterates just once over the corporations:

```suggestion
        if (@diff_corp_sizes = @game.all_corporations.any? { |c| @game.corporation_size(c) != :small })
```

**Comment by yzemaze** [comment]
@Zwergenpunk 
NB https://github.com/tobymao/18xx/pull/4468/commits/4d2508f0be8238605e5a8d75fb9b326b5186c23a if you’d still like to replace the long strings on corp cards with their abbreviations.

**Review by tobymao — APPROVED** [review]
